Source/TestDafny/MultiBackendTest.cs

```diff
@@ -339,15 +357,70 @@ int ExpectedExitCode
       dafnyArgs = dafnyArgs.Concat(new[] { "--include-runtime:false", "--input", runtimePath });
     }
 
-    var (exitCode, outputString, error) = await RunDafny(options.DafnyCliPath, dafnyArgs);
+    int exitCode;
+    string outputString;
+    string error;
+    try {
+      (exitCode, outputString, error) = await RunDafny(options.DafnyCliPath, dafnyArgs);
+    } catch (Exception e) {
+      (exitCode, outputString, error) = (3, e.ToString(), e.ToString());
+    }
+
     var compilationOutputPrior = new Regex("\r?\nDafny program verifier[^\r\n]*\r?\n").Match(outputString);
     if (compilationOutputPrior.Success) {
       outputString = outputString.Remove(0, compilationOutputPrior.Index + compilationOutputPrior.Length);
     }
 
+    var diffMessage = exitCode == 0 ? AssertWithDiff.GetDiffMessage(expectedOutput, outputString) : null;
+    if (checkFile == null && exitCode != 0) {
+      if (Environment.GetEnvironmentVariable("DAFNY_INTEGRATION_TESTS_UPDATE_TARGET_EXPECT_FILE") == "true") {
+        if ((Environment.GetEnvironmentVariable("DAFNY_INTEGRATION_TESTS_ROOT_DIR") ?? "") == "") {
+          await output.WriteLineAsync(
+            "DAFNY_INTEGRATION_TESTS_UPDATE_TARGET_EXPECT_FILE is true but DAFNY_INTEGRATION_TESTS_ROOT_DIR is not set");
+        } else {
+          // Ensure we don't accidentally do this for every compiler.
```

Comment:
Why not? See also my comment to make a single `DAFNY_INTEGRATION_TESTS_UPDATE_EXPECT_FILES` environment variable below.